infra/modules/worklytics-connector-specs/main.tf

```diff
@@ -383,6 +383,8 @@ EOT
         GRANT_TYPE : "client_credentials"
         CREDENTIALS_FLOW : "client_secret"
         REFRESH_ENDPOINT : "https://${var.salesforce_domain}/services/oauth2/token"
+        ACCESS_TOKEN_CACHEABLE : "true",
```

Comment:
so this doesn't make a difference in any case, right?
Tokens can be locally cacheable **always**, if the container allows it, it will benefit from it. Otherwise, use the shared token is the way to go.
There is no way to have one token "locally" per function instance... So probably I'll just revert the ACCESS_TOKEN_CACHEABLE, I think in the end it just add noise and another config that won't matter much

Reply:
in theory, there should be. caching doesn't seem to work in GCP as the instance of `RefreshHandlerImpl` it's cached on is getting lost between requests.

if we get the object graph fixed to persist that across requests, this should fully work as expected and caching tokens should work independently of sharing tokens.

caching tokens should really be the default, yes; Zoom is the exception that has some kind of strict token rotation policy (can't find a term for this).

sharing could be the default, at the risk of cycling load on Secret Manager / Parameter Store, and creating lots of versions of the token as we saw with Zoom.